### ISSUE TITLE
WritableView Implementation

### ISSUE THREAD

**Issue by zerodev1200** [Cysharp/ObservableCollections#75]
# Issues with WritableView Implementation

## 1. Environment
- Verification performed in WPF

### Issue Description
- Double-clicking on the empty row at the bottom of a bound DataGrid triggers an exception in `NotifyCollectionChangedSynchronizedViewList`
```csharp
public int Add(object? value)
{
    throw new NotImplementedException();
}
```

### Additional Observations
- No exception occurs when editing cells with existing values
- Editing functionality was working in versions up to v3.0.4
- Hypothesis: In WPF, editing operations alone don't pass through `object? IList.this[int index]`

## 3. Unexpected Behavior with WritableView Filter
Added the following code to `ObservableCollections.sandbox.ConsoleApp`
### Reproduction Code
```csharp
notify[1] = "99999";
view.AttachFilter(x => x % 2 == 0);
Console.WriteLine("view");
foreach (var item in view)
{
    Console.WriteLine(item);
}
Console.WriteLine("");
Console.WriteLine("l(source)");
foreach (var item in l)
{
    Console.WriteLine(item);
}
```

### Current Output
```
view
0
99999
4

l(source)
0
99999
2
3
4
5
```

### Expected Output
```
view
0
4

l(source)
0
1
99999
3
4
5
```

## 4. Related Improvements
- I have submitted a pull request addressing the overloads of ToWritableNotifyCollectionChanged. #74

**Comment by neuecc** [comment]
Thank you, I've released 3.1.1, that fixed these issues.

2: I've implement Add operation for WritableNotifyCollectionChanged

Usage example:
```csharp
IList<string?> bindable = view.ToWritableNotifyCollectionChanged((string? newView, Person original, ref bool setValue) =>
{
    if (setValue)
    {
        // default setValue == true is Set operation
        original.Name = newView;

        // You can modify setValue to false, it does not set original collection to new value.
        // For mutable reference types, when there is only a single,
        // bound View and to avoid recreating the View, setting false is effective.
        // Otherwise, keeping it true will set the value in the original collection as well,
        // and change notifications will be sent to lower-level Views(the delegate for View generation will also be called anew).
        setValue = false;
        return original;
    }
    else
    {
        // default setValue == false is Add operation
        return new Person { Age = null, Name = newView };
    }
});
```

3: It is bug! I've fixed!

**Comment by zerodev1200** [comment]
I may have seen a glimmer of hope.
I'll investigate a bit more and submit a PR...!
